src/Components/Web/src/WebEventData.cs

```diff
@@ -3,57 +3,57 @@
 
 using System;
 using System.Text.Json;
-using System.Threading.Tasks;
 using Microsoft.AspNetCore.Components.Rendering;
-using Microsoft.JSInterop;
 
 namespace Microsoft.AspNetCore.Components.Web
 {
     /// <summary>
-    /// Provides mechanisms for dispatching events to components in a <see cref="Renderer"/>.
+    /// For framework use only.
     /// </summary>
-    public static class RendererRegistryEventDispatcher
+    public class WebEventData
     {
+        // This class represents the second half of parsing incoming event data,
+        // once the type of the eventArgs becomes known.
+
```

Comment:
The more I look at this, the more I think we should just make the deserialization piece shared source and internal. That way the public API surface is not polluted and we are free to change the implementation in the future if we choose to do so.

Reply:
I agree the pubternal is awkward, but remembering ComponentsElectron, I think we have a reason to make this a regular `public` supported type.

Reply:
Could we make that decision later? We can always make things public, we can't do otherwise. For ComponentsElectron there might be other options like creating a SourcesPackage if we have to.

Reply:
If we have to, we can copy a bit of code to componentselectron, create a new package or expose a new type when we have to, but we are not shipping that right now, so I would like us to avoid making a decision we might regret later.

Reply:
OK, that sounds fine. It's not something I feel desperately strongly about, and I agree there are possible workarounds for ComponentsElectron.

I'm making `WebEventData` (with all the parsing logic) internal and a shared-source-file.

The pure-DTO `WebEventDescriptor` class needs to be `public` though because it's part of the method signature for a `[JSInvokable]` method for client-side Blazor. Technically we could do some obscure thing with `ifdef`s to make it shared-source and internal for `.Components` and public for `.Blazor`, but since it's now purely a DTO with no behavior, I'm not worried about it being part of the public API.